include/istio/control/http/check_data.h

```diff
@@ -38,8 +38,8 @@ class CheckData {
   // Get downstream tcp connection ip and port.
   virtual bool GetSourceIpPort(std::string *ip, int *port) const = 0;
 
-  // If SSL is used, get origin user name.
-  virtual bool GetSourceUser(std::string *user) const = 0;
+  // If SSL is used, get certificate SAN URI.
+  virtual bool GetPrincipal(bool peer, std::string *user) const = 0;
```

Comment:
Wouldn't it be better to have different methods to be more explicit and clear, instead of using an argument to control it?

Reply:
Well, @qiwzhang thinks it's more valuable to share the code between the two functions, which minimizes new code additions.

Reply:
FWIW, sharing the function eliminated 40 extra lines.